app/forms/providers/means/regular_outgoings_form.rb

```diff
@@ -0,0 +1,161 @@
+
+      OUTGOING_TYPES.each do |outgoing_type|
+        attr_accessor "#{outgoing_type}_amount".to_sym, "#{outgoing_type}_frequency".to_sym
+      end
+
+      validates :transaction_type_ids, presence: true, unless: :none_selected?
+      validate :all_regular_transactions_valid
+
+      def initialize(params = {})
+        @none_selected = none_selected.in?(params["transaction_type_ids"] || [])
+        @legal_aid_application = params.delete(:legal_aid_application)
+        @transaction_type_ids = params["transaction_type_ids"] || @legal_aid_application.transaction_type_ids
+
+        assign_regular_transaction_attributes
+
+        super
+      end
+
+      def save
+        return false unless valid?
+
+        ApplicationRecord.transaction do
+          destroy_transactions!
+
+          build_legal_aid_application_transaction_types
+          legal_aid_application.assign_attributes(
+            no_debit_transaction_types_selected: none_selected?,
+          )
+          legal_aid_application.save!
+
+          regular_transactions.each(&:save!)
+        end
+
+        true
+      end
+
+      def transaction_type_ids=(ids)
+        @transaction_type_ids = if none_selected?
+                                  []
+                                else
+                                  ids.compact_blank
+                                end
+      end
+
+      def transaction_type_options
+        TransactionType.not_children.debits
+      end
+
+      def none_selected
+        "none"
+      end
+
+    private
+
+      def assign_regular_transaction_attributes
+        regular_transactions.each do |transaction|
+          transaction_type = transaction.transaction_type
+          public_send("#{transaction_type.name}_amount=", transaction.amount)
+          public_send("#{transaction_type.name}_frequency=", transaction.frequency)
+        end
+      end
+
+      def none_selected?
+        @none_selected
+      end
+
+      def transaction_types
+        transaction_type_options.where(id: transaction_type_ids)
+      end
+
+      def destroy_transactions!
+        destroy_legal_aid_application_transaction_types!
+        destroy_regular_outgoing_transactions!
+        destroy_cash_outgoing_transactions!
+      end
+
+      def destroy_legal_aid_application_transaction_types!
+        legal_aid_application
+          .legal_aid_application_transaction_types
+          .debits
+          .where.not(transaction_type_id: transaction_type_ids)
+          .destroy_all
+      end
+
+      def destroy_regular_outgoing_transactions!
+        legal_aid_application
+          .regular_outgoings
+          .without(existing_outgoing_regular_transactions)
+          .destroy_all
+      end
+
+      def destroy_cash_outgoing_transactions!
+        legal_aid_application
+          .cash_transactions
+          .debits
+          .where.not(transaction_type_id: transaction_type_ids)
+          .destroy_all
```

Comment:
Again, I realize this copies the pattern that went in for regular incomes, so maybe @cpjmcquillan can comment, but I do not believe this is needed as the `destroy_legal_aid_application_transaction_types!` method destroys the join table record which will fire this [after_commit hook](https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/main/app/models/legal_aid_application_transaction_type.rb#L8), which will delete the cash_transactions.

by extension, the same could be done for regular transactions, to take the dependent destroy logic out of the controller and into the model, where it feels to me to be more appropriate.

Anyway, neither of there are blockers and should not really be done here as the pattern is already in but it would be good could look at this as a refactor I think.

Reply:
This is intentional.

It only isn't needed if the `TransactionType` record associated to the `{Regular/Cash/Bank}Transaction` record has a respective `LegalAidApplicationTransactionType` record. 

This relationship isn't enforced at the database or the application layer, though —so it's possible for records to exist that aren't destroyed by the callback (because the callback never happens).

I think it's safer to be intentional about destroying the records until we refactor the relationship between a `LegalAidApplication` and it's various transactions (i.e they are joined through the join table).

If we do that, we wouldn't need the callback method anymore either, since we could rely on `dependent: :destroy` to do the dirty work for us.